contracts/BC_fusion/GovToken.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20BurnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20PermitUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20VotesUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "./System.sol";
+
+interface IStakePool {
+    function getPooledBNB(address account) external view returns(uint256);
+}
+
+contract GovToken is
+    System,
+    Initializable,
+    ERC20Upgradeable,
+    ERC20BurnableUpgradeable,
+    ERC20PermitUpgradeable,
+    ERC20VotesUpgradeable
+{
+    string private constant NAME = "BSC Governance Token";
+    string private constant SYMBOL = "govBNB";
+
+    // validator StakePool contract => user => amount
+    mapping(address => mapping(address => uint256)) public mintedMap;
+
+    function initialize()
+    public
+    initializer
+    onlyCoinbase
+    onlyZeroGasPrice
+    {
+        __ERC20_init(NAME, SYMBOL);
+        __ERC20Burnable_init();
+        __ERC20Permit_init(NAME);
+        __ERC20Votes_init();
+    }
+
+    function sync(address[] calldata validatorPools, address account) external onlyStakeHub {
+        uint256 _length = validatorPools.length;
+        for (uint256 i = 0; i < _length; ++i) {
+            _sync(validatorPools[i], account);
+        }
+    }
+
+    function delegateVote(address delegator, address delegatee) external onlyStakeHub {
```

Comment:
Will “tally” use the user's account to call this function directly?   if yes, why it is not a inheritance function. 

And the access control `onlyStakeHub` is wrong to me.

Reply:
"tally" use `delegate` method exist in the GovToken contract(inherited from `ERC20VotesUpgradeable`). `delegateVote` is an additional interface that makes it easy for `StakeHub` to do some bundle operations.